src/main/java/pl/scoutCamp/business/RankingService.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.data.domain.PageImpl;
+import org.springframework.stereotype.Service;
+import pl.scoutCamp.api.controller.enums.SortOrder;
+import pl.scoutCamp.api.controller.enums.SortType;
+import pl.scoutCamp.business.dao.TeamCategorizationSheetDAO;
+import pl.scoutCamp.business.dao.TeamDAO;
+import pl.scoutCamp.domain.Ranking;
+import pl.scoutCamp.domain.Team;
+import pl.scoutCamp.domain.TeamCategorizationSheet;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static pl.scoutCamp.api.controller.enums.GivenCategory.*;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class RankingService {
+
+    private final TeamDAO teamDAO;
+    private final TeamCategorizationSheetDAO teamCategorizationSheetDAO;
+
+    public Page<Ranking> createRankingList(
+            String period,
+           SortType sort,
+            SortOrder order) {
+        List<Ranking> rankingList = new ArrayList<>();
+        for (Team team : findAllTeams()) {
+            Integer totalPoints = getTotalPoints(getTeamSheets(team, period));
+            Ranking teamSummary = Ranking.builder()
+                    .teamName(team.getName())
+                    .totalPoints(totalPoints)
+                    .category(getCategory(totalPoints))
+                    .sheets(getTeamSheets(team, period))
+                    .build();
+            rankingList.add(teamSummary);
+        }
+//
+        return new PageImpl<>(sortedRankingList(rankingList, sort, order));
+    }
+
+    private List<Ranking> sortedRankingList(List<Ranking> rankingList, SortType sort, SortOrder order) {
+        Comparator<Ranking> comparingByPoints = Comparator.comparing(Ranking::getTotalPoints);
+        Comparator<Ranking> comparingByName = Comparator.comparing(Ranking::getTeamName);
+        Comparator<Ranking> comparingByCategory = Comparator.comparing(Ranking::getCategory);
+        if (order.equals(SortOrder.ASC)) {
+            switch (sort) {
+                case POINTS -> compareAsc(rankingList, comparingByPoints);
+                case NAME -> compareAsc(rankingList, comparingByName);
+                case CATEGORY -> compareAsc(rankingList, comparingByCategory);
+            }
+        } else {
+            switch (sort) {
+                case POINTS -> compareDesc(rankingList, comparingByPoints);
+                case NAME -> compareDesc(rankingList, comparingByName);
+                case CATEGORY -> compareDesc(rankingList, comparingByCategory);
+            }
+        }
+        return rankingList;
+    }
+
+    private static void compareAsc(List<Ranking> rankingList, Comparator<Ranking> comparator) {
+        rankingList.sort(comparator);
+    }
+    private static void compareDesc(List<Ranking> rankingList, Comparator<Ranking> comparator) {
+        rankingList.sort(comparator.reversed());
+    }
+
+    private String getCategory(int points) {
+        String category = "";
+        if (points >= 0 && points < 20) {
```

Comment:
I've missed that in the previous CR. We shouldn't place "magic numbers" in the code like there. It's better to move them into some global variables with meaningful names, like "private static Integer MIN_POINTS_FOR_POLOWA = 20" etc.

Reply:
Reference: https://medium.com/@samaddico/programming-and-magic-numbers-f766e0cd1369